src/filters/dasher.c

```diff
@@ -4729,11 +4729,11 @@ static void dasher_purge_segments(GF_DasherCtx *ctx, u64 *period_dur)
 				evt.file_del.url = sctx->filepath;
 				gf_filter_pid_send_event(ds->opid, &evt);
 				//purge LLHLS frags
-				if (sctx->frags && (ctx->llhls>1)) {
+				if (sctx->frags && (ctx->llhls>1 || ds->set->ssr_mode)) {
 					u32 k;
 					for (k=0; k<sctx->nb_frags; k++) {
 						char szTmp[100];
-						sprintf(szTmp, ".%u", k+1);
+						sprintf(szTmp, ".%u", k + (ds->set->ssr_mode ? 0 : 1));
```

Comment:
Currently HLS in test mode uses 1-based index and 0-based index otherwise, see https://github.com/gpac/gpac/blob/master/src/media_tools/mpd.c#L4016.
So until we fix this this, I believe the proper patch is

```
sprintf(szTmp, ".%u", k + (ds->set->ssr_mode || !gf_sys_is_test_mode() ? 0 : 1));
```

Reply:
also the same test needs to be done here https://github.com/gpac/gpac/blob/master/src/filters/dasher.c#L10602